### PR TITLE
Add overload for private key authentication on sftp configuration builder

### PR DESCRIPTION
**Pull request by IamMartinPeek**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**: related to: it allows private key file authentication for sftp configuration also when keyfile has no password

Please reference the issue this PR will close: #_[1833]_

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**: not really it just extends functionallity

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [ ] Created/updated tests
- [ ] Unit tests passing
- [ ] End-to-end tests passing
- [ ] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

**Comment by IamMartinPeek** [comment]
@dotnet-policy-service agree

File: src/HealthChecks.Network/SftpConfiguration.cs [hunk 1]
```diff
@@ -44,6 +44,12 @@ public SftpConfigurationBuilder AddPrivateKeyAuthentication(string privateKey, s
 
         return this;
     }
+    public SftpConfigurationBuilder AddPrivateKeyAuthentication(PrivateKeyFile privateKey)
+    {
+        AuthenticationMethods.Add(new PrivateKeyAuthenticationMethod(_userName, privateKey));
+        
+        return this;
+    }
```
Comment (sungam3r) [inline src/HealthChecks.Network/SftpConfiguration.cs:52]:
```suggestion

    public SftpConfigurationBuilder AddPrivateKeyAuthentication(PrivateKeyFile privateKey)
    {
        AuthenticationMethods.Add(new PrivateKeyAuthenticationMethod(_userName, privateKey));
        
        return this;
    }

```

Comment (sungam3r) [reply src/HealthChecks.Network/SftpConfiguration.cs:53]: to trigger CI

**Comment by sungam3r** [comment]
@IamMartinPeek Thanks. I'm fine to merge after updating API approval file.

File: src/HealthChecks.Network/SftpConfiguration.cs [hunk 2]
```diff
@@ -44,6 +44,14 @@ public SftpConfigurationBuilder AddPrivateKeyAuthentication(string privateKey, s
 
         return this;
     }
+
+    public SftpConfigurationBuilder AddPrivateKeyAuthentication(PrivateKeyFile privateKey)
+    {
+        AuthenticationMethods.Add(new PrivateKeyAuthenticationMethod(_userName, privateKey));
+
```
Comment (sungam3r) [inline src/HealthChecks.Network/SftpConfiguration.cs:51]:
```suggestion
        AuthenticationMethods.Add(new PrivateKeyAuthenticationMethod(_userName, privateKey));

```

**Comment by sungam3r** [comment]
Ah, CI is disabled, see #1018 . OK, I merge this as is.

**Pull request merged**